src/compiler/crystal/semantic/bindings.cr

```diff
@@ -415,8 +415,8 @@ module Crystal
       return unless self.def.args.all? &.type?
       return unless self.def.type?
 
-      types = self.def.args.map &.type
-      return_type = @force_nil ? self.def.type.program.nil : self.def.type
+      types = self.def.args.map &.type.virtual_type
```

Comment:
the spec is not asserting virtual_type is used in arguments

Reply:
Good catch. I'll update the spec tomorrow.

Reply:
Done!